flink-runtime/src/main/java/org/apache/flink/runtime/io/network/netty/PartitionRequestQueue.java

```diff
@@ -61,6 +62,9 @@
 	/** The readers which are already enqueued available for transferring data. */
 	private final ArrayDeque<NetworkSequenceViewReader> availableReaders = new ArrayDeque<>();
 
+	/** The readers are marked as to be closed, waiting for confirmation from writer side */
+	private final Set<NetworkSequenceViewReader> readersToClose = new HashSet<>();
```

Comment:
This set should be cleared in failover cases such as channel inactive and exception caught. You can refer to the variable `availableReaders`